mountpoint-s3/src/cli.rs

```diff
@@ -877,6 +877,12 @@ where
 
     let (client, runtime, s3_personality) = client_builder(&args)?;
 
+    if args.incremental_upload && !matches!(s3_personality, S3Personality::ExpressOneZone) {
+        return Err(anyhow!(
+            "--incremental-upload is only supported for S3 Express One Zone buckets"
```

Comment:
This is what we use in [docs](https://github.com/awslabs/mountpoint-s3/blob/main/doc/SEMANTICS.md#reading-and-writing-files):
```suggestion
            "--incremental-upload is only supported on directory buckets in S3 Express One Zone"
```